### PR TITLE
Adds 7 new site coverages

### LINKED ISSUE #1145: peelwithzeal coverage
Please check that recipes published on the website you're requesting are public (we can't currently scrape recipes that require an account login), and add sample recipe URL(s) below:

- https://www.peelwithzeal.com/molasses-baked-beans/
- https://www.peelwithzeal.com/cajun-crab-dip/

Can support ingredient_groups and equipment. Will implement myself.

### PR DESCRIPTION
**Pull request by jknndy**
All schema based with `ingredient_grouping` coverage & 1 with equipment

Resolves #1145

### CODE REVIEW

**Comment by jknndy** [comment]
Not sure what is causing the issue & im unable to replicate it locally. Some of the unittests are passing & others failing on the readme

File: recipe_scrapers/ingoodflavor.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# mypy: allow-untyped-defs
```
Comment (jayaddison) [inline recipe_scrapers/ingoodflavor.py:1]: @jknndy these `allow-untyped-defs` directives can be removed now (credit: #1174)

File: tests/test_data/ingoodflavor.com/ingoodflavor_1.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        "1 tbsp. extra virgin olive oil",
+        "2 cups chopped onions",
+        "2 cloves garlic, minced",
+        "2 (about 14 -16 oz each) packages ground Portuguese linguica",
+        "4 (15 oz.) cans sliced potatoes, drained well and coarsely chopped",
+        "3/4 tsp. Lawry’s Seasoned Salt",
+        "1/2 tsp. black pepper",
+        "10 oz. (about 5 cups) shredded sharp cheddar cheese"
+      ],
+      "purpose": "Ingredients"
```
Comment (jayaddison) [inline tests/test_data/ingoodflavor.com/ingoodflavor_1.json:30]: Nitpick: this `purpose` field seems redundant - I think we'd generally expect/want `null` on a single-group recipe.

File: tests/test_data/kitchendreaming.com/kitchendreaming_1.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
+  "author": "Ronda Eagle | Kitchen Dreaming",
+  "canonical_url": "https://kitchendreaming.com/crock-pot-buffalo-chicken-meatballs/",
+  "site_name": "KitchenDreaming",
+  "host": "kitchendreaming.com",
+  "language": "en-US",
+  "title": "Crock Pot Buffalo Chicken Meatballs",
+  "ingredients": [
+    "1-1/2 pounds ground chicken ([See Note 1])",
+    "1/2 cup panko bread crumbs ( [See Note 2])",
```
Comment (jayaddison) [inline tests/test_data/kitchendreaming.com/kitchendreaming_1.json:10]: For a moment I was wondering whether to suggest attempting to filter out these `See Note` statements -- we don't include the accompanying notes in the scraper field output, as far as I know.  However: I think we should keep them.  Removing them would mean that users would be entirely unaware that there is additional (and presumably, somewhat relevant -- if it is called out in the ingredient list) context about them.

They may have to seek that information out themselves if they want/need to know it - but I think that's a better tradeoff than hiding it entirely.

(in a philosophical sense someone might argue that what scrapers do in their entirety is to remove surrounding context -- I'd argue that there's some truth in that, but that it's possible to make judgement calls about what is relevant and what is not)

File: tests/test_data/kitchendreaming.com/kitchendreaming_1.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  ],
+  "category": "Appetizer",
+  "yields": "20 servings",
+  "description": "Buffalo Chicken wings are a favorite American appetizer or meal. So when I decided to turn them into a fun party snack, I knew they would be an instant hit.",
+  "total_time": 130,
+  "cook_time": 120,
+  "prep_time": 10,
+  "cuisine": "American",
+  "ratings": 5.0,
+  "ratings_count": 1.0,
```
Comment (jayaddison) [inline tests/test_data/kitchendreaming.com/kitchendreaming_1.json:62]:
```suggestion
  "ratings_count": 1,
```

File: tests/test_data/kitchendreaming.com/kitchendreaming_2.json [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    "Garnish with chopped cilantro and serve with rice and Naan bread."
+  ],
+  "category": "Dinner,Main,Main Course,main meal",
+  "yields": "8 servings",
+  "description": "Indian Butter Chicken ~ also known as Chicken Makhani, is a classic, flavorful Indian dish. In this easy, make-at-home version of the restaurant favorite, you control the heat. Make it as spicy or mild as you want! This Butter Chicken recipe uses flavorful spices like garam masala, coriander, & turmeric mixed with tender chicken, cream, butter & tomatoes. There's plenty of that mild curry sauce to soak up with naan or rice! Simply marinade the chicken, and then this dish is ready in 30 minutes!",
+  "total_time": 30,
+  "cook_time": 30,
+  "cuisine": "Indian",
+  "ratings": 5.0,
+  "ratings_count": 1.0,
```
Comment (jayaddison) [inline tests/test_data/kitchendreaming.com/kitchendreaming_2.json:86]:
```suggestion
  "ratings_count": 1,
```

File: tests/test_data/leitesculinaria.com/leitesculinaria_1.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    "Taste and adjust the seasoning and consistency of the final sauce, adding additional vinegar, water, salt, granulated sugar, or garlic powder to suit your taste. Transfer to a glass jar, close the lid tightly, and refrigerate for up to 6 months."
+  ],
+  "category": "Condiments",
+  "yields": "16 servings",
+  "description": "This homemade Sriracha sauce, made with everyday ingredients including hot peppers, vinegar, garlic, and salt, is easy to make, incendiary in taste, and less salty than the traditional version.",
+  "total_time": 10080,
+  "prep_time": 5,
+  "cuisine": "Thai",
+  "ratings": 4.85,
+  "ratings_count": 63.0,
```
Comment (jayaddison) [inline tests/test_data/leitesculinaria.com/leitesculinaria_1.json:55]:
```suggestion
  "ratings_count": 63,
```

File: tests/test_data/leitesculinaria.com/leitesculinaria_2.json [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  ],
+  "category": "Mains",
+  "yields": "8 servings",
+  "description": "Caçoila, or Portuguese stewed beef, is a comforting braise of beef, red peppers, and aromatics, cooked very slowly in a red wine, tomato, and butter sauce.",
+  "total_time": 1440,
+  "cook_time": 180,
+  "prep_time": 20,
+  "cuisine": "Azorean",
+  "ratings": 4.88,
+  "ratings_count": 24.0,
```
Comment (jayaddison) [inline tests/test_data/leitesculinaria.com/leitesculinaria_2.json:75]:
```suggestion
  "ratings_count": 24,
```

**Comment by jayaddison** [comment]
A few nitpicks, but generally these look good to me, thanks @jknndy - I'll add another round of review after you have time to respond.

File: tests/test_data/ingoodflavor.com/ingoodflavor_2.json [hunk 8]
```diff
@@ -0,0 +1,83 @@
+{
+  "author": "In Good Flavor",
```
Comment (jayaddison) [inline tests/test_data/ingoodflavor.com/ingoodflavor_2.json:2]: We could be slightly more precise here, by gathering  `Thao | In Good Flavor` from the author info on the page.

**Comment by jknndy** [comment]
All caught!

File: README.rst [hunk 9]
```diff
@@ -311,6 +316,7 @@ Scrapers available for:
 - `https://www.panelinha.com.br/ <https://www.panelinha.com.br>`_
 - `https://paninihappy.com/ <https://paninihappy.com>`_
 - `https://www.persnicketyplates.com/ <https://www.persnicketyplates.com/>`_
+- `https://www.peelwithzeal.com/ <https://www.peelwithzeal.com/>`_
```
Comment (jayaddison) [inline README.rst:319]:
```suggestion
- `https://www.peelwithzeal.com/ <https://www.peelwithzeal.com/>`_
- `https://www.persnicketyplates.com/ <https://www.persnicketyplates.com/>`_
```

Should do it I think :)

**Comment by jayaddison** [comment]
It seems strange that the readme-check unit test passed on some versions of Python and not others -- I'll start investigating that soon.

**Comment by jayaddison** [comment]
> It seems strange that the readme-check unit test passed on some versions of Python and not others -- I'll start investigating that soon.

Ah, I forgot: the reason is that the fix to read the `README` contents from the package metadata (#1162) uses a mechanism that is only available on Py3.10+ and skips the test for other versions.

At the time I felt that that was a reasonable tradeoff -- but for consistency, and so that me and others don't forget this again and get confused by it, it might be worthwhile to provide an implementation for the older versions of Python.

**Review by jayaddison — APPROVED** [review]

**Pull request merged**